docs/make.jl

```diff
@@ -0,0 +1,80 @@
+#
+# With minor changes from https://github.com/JuliaGaussianProcesses/AbstractGPs.jl/docs
+#
+### Process examples
+# Always rerun examples
+const EXAMPLES_OUT = joinpath(@__DIR__, "src", "examples")
+ispath(EXAMPLES_OUT) && rm(EXAMPLES_OUT; recursive=true)
+mkpath(EXAMPLES_OUT)
+
+# Install and precompile all packages
+# Workaround for https://github.com/JuliaLang/Pkg.jl/issues/2219
+examples = filter!(isdir, readdir(joinpath(@__DIR__, "..", "examples"); join=true))
+let script = "using Pkg; Pkg.activate(ARGS[1]); Pkg.instantiate()"
+    for example in examples
+        if !success(`$(Base.julia_cmd()) -e $script $example`)
+            error(
+                "project environment of example ",
+                basename(example),
+                " could not be instantiated",
+            )
+        end
+    end
+end
+# Run examples asynchronously
+processes = let literatejl = joinpath(@__DIR__, "literate.jl")
+    map(examples) do example
+        return run(
+            pipeline(
+                `$(Base.julia_cmd()) $literatejl $(basename(example)) $EXAMPLES_OUT`;
+                stdin=devnull,
+                stdout=devnull,
+                stderr=stderr,
+            );
+            wait=false,
+        )::Base.Process
+    end
+end
+
+# Check that all examples were run successfully
+isempty(processes) || success(processes) || error("some examples were not run successfully")
+
+# Building Documenter
+using Documenter
+
+# Print `@debug` statements (https://github.com/JuliaDocs/Documenter.jl/issues/955)
+if haskey(ENV, "GITHUB_ACTIONS")
+    ENV["JULIA_DEBUG"] = "Documenter"
+end
+
+using AdvancedPS
+
+DocMeta.setdocmeta!(AdvancedPS, :DocTestSetup, :(using AdvancedPS); recursive=true)
+
+makedocs(;
+    sitename="AdvancedPS",
+    format=Documenter.HTML(),
+    modules=[AdvancedPS],
+    pages=[
+        "Home" => "index.md",
+        "api.md",
+        "Examples" => [
+            "example.md",
+            map(
+                (x) -> joinpath("examples", x),
+                filter!(filename -> endswith(filename, ".md"), readdir(EXAMPLES_OUT)),
+            )...,
+        ],
+    ],
+    checkdocs=:exports,
```

Comment:
Maybe we could use
```suggestion
    strict=true,
    checkdocs=:exports,
```
to ensure that an error is thrown and the action fails if something fails or e.g. is not documented. Otherwise one always has to check the logs.

Reply:
Ah, missed that. Yes let's fail instead